nntrainer/layers/pooling2d_layer.cpp

```diff
@@ -69,25 +84,27 @@ void Pooling2DLayer::forwarding(bool training) {
     hidden_ = Tensor(hidden_dim);
   }
 
-  if (pooling_type == PoolingType::max) {
-    max_idx.resize(output_dim[0].getDataLen());
-  } else if (pooling_type == PoolingType::global_max) {
-    max_idx_global.resize(output_dim[0].getDataLen());
+  if (training) {
+    if (pooling_type == PoolingType::global_max) {
+      max_idx_global.clear();
```

Comment:
move this inside `pooling2d` where this vector will be reserved again.

Reply:
.clear() is guaranteed to keep it's capacity so it does not need to reserve. let's discuss this offline.

Reply:
This is not critical so let's have a discussion after merge.